### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by cristovaoth**
This branch renames some of the setter and checker functions, most notably:
* setCanDelegateToTarget
* setCanSendToTarget
* allowTarget
* allowFunction
* scopeFunction

There is also a new function called "scopeParameter", which allows an owner to scope one single parameter (as opposed to all with scopeFunction).

In addition to the already existing Roles.spec.ts which is function oriented, I added new test files which are use case oriented - Comparison, Scoping and Clearance

### CODE REVIEW

File: contracts/Roles.sol [hunk 1]
```diff
@@ -126,126 +124,165 @@ contract Roles is Modifier {
...
     ) external onlyOwner {
-        Permissions.setParametersScoped(
-            roleList,
-            role,
-            targetAddress,
-            functionSig,
-            paramsScoped,
-            types,
-            compTypes
+        // 24kb
```
Comment (auryn-macmillan) [inline contracts/Roles.sol:212]: Did you mean to leave this chunk of commented out code?

File: contracts/Roles.sol [hunk 2]
```diff
@@ -126,126 +124,165 @@ contract Roles is Modifier {
...
+        //     isParamScoped.length == isParamDynamic.length,
+        //     "Mismatch: isParamScoped and isParamDynamic length"
+        // );
+
+        // require(
+        //     isParamScoped.length == paramCompType.length,
+        //     "Mismatch: isParamScoped and paramCompType length"
+        // );
+
+        require(
```
Comment (auryn-macmillan) [inline contracts/Roles.sol:223]: Would prefer to use a custom error.

**Review by auryn-macmillan — CHANGES_REQUESTED** [review]
Changes mostly look great!

Two changes:
1. Please remove commented out code.
2. Let's use custom errors rather than require statements.

File: contracts/Roles.sol (hunk 1 above)
Comment (cristovaoth) [reply contracts/Roles.sol:212]: Left it, since this was not a final branch. Will push the removal.

File: contracts/Roles.sol (hunk 2 above)
Comment (cristovaoth) [reply contracts/Roles.sol]: Done

**Pull request merged**